### PR TITLE
PdfComparator with base64

### PR DESCRIPTION
**Pull request by LeJeanbono**
What do yout think @finsterwalder ?
I will update the Readme if it's fine to you.

### CODE REVIEW

**Comment by finsterwalder** [comment]
I don't like changing the public API of the service, so I would rather try to find a way that leaves the old API intact.

**Comment by finsterwalder** [comment]
readAllBytes on InputStream is available in Java 9, but the lib is compatible with Java8.
You code has a copy-paste-bug (see comments in code).

File: src/main/java/de/redsix/pdfcompare/PdfComparator.java [hunk 1]
```diff
@@ -70,17 +72,25 @@ private PdfComparator(T compareResult) {
...
+        if (!expectedPdf.equals(actualPdf)) {
+            if (expectedPdf.endsWith(".pdf")) {
+                this.expectedStreamSupplier = () -> Files.newInputStream(Paths.get(expectedPdf));
+            } else {
+                this.expectedStreamSupplier = () -> new ByteArrayInputStream(Base64.getDecoder().decode(expectedPdf));
+            }
+            if (actualPdf.endsWith(".pdf")) {
+                this.actualStreamSupplier = () -> Files.newInputStream(Paths.get(actualPdf));
+            } else {
+                this.expectedStreamSupplier = () -> new ByteArrayInputStream(Base64.getDecoder().decode(actualPdf));
```
Comment (finsterwalder) [inline src/main/java/de/redsix/pdfcompare/PdfComparator.java:92]: this should be this.actualStreamSupplier

File: src/test/java/de/redsix/pdfcompare/IntegrationTest.java [hunk 2]
```diff
@@ -209,6 +210,27 @@ public void identicalFilenamesAreEqual() throws IOException {
         writeAndCompare(result);
     }
 
+    @Test
+    public void compareWithBase64() throws IOException {
+        String expectedBase64 = Base64.getEncoder().encodeToString(toByteArray(r("expected.pdf")));
+        String actualBase64 = Base64.getEncoder().encodeToString(toByteArray(r("actual.pdf")));
+        final CompareResult result = new PdfComparator(expectedBase64, actualBase64).compare();
+        assertThat(result.isEqual(), is(true));
```
Comment (finsterwalder) [inline src/test/java/de/redsix/pdfcompare/IntegrationTest.java:218]: this should be false, since the expected.pdf and the actual.pdf files are note equal

Comment (LeJeanbono) [reply src/test/java/de/redsix/pdfcompare/IntegrationTest.java:218]: Oh yes, sorry...

**Comment by finsterwalder** [comment]
I created a different solution in version 1.1.50.
That allows to compare PDFs in base64 encoding by using a static
PdfComparator.base64(expectedPdf, actualPdf)-Method
Thanks for the idea to use base64 encoded strings.

**Comment by LeJeanbono** [comment]
Thanks @finsterwalder 
The best would be to mix different inputs => Path & fileName, Base64 & File

**Comment by finsterwalder** [comment]
Nice idea. I'm thinking of a builder-like API. That would allow for that kind of flexibility.
I will think about that...

**Pull request closed without merging**